src/Combinatorics/Matroids/realization_space.jl

```diff
@@ -0,0 +1,989 @@
+        else
+            print(io, "The matroid is not realizable over the specified field or characteristic.")
+        end
+    else
+        if RS.one_realization
+            println(io, "One realization is given by")
+        else
+            println(io, "The realizations are parametrized by")
+        end
+        # println isn't ideal as it prints the matrix as one big line
+        display(RS.realization_matrix)
+        println(io, "in the ", RS.ambient_ring)
+        I = RS.defining_ideal
+        if (typeof(I) <: NumFieldOrdIdl && I.gen != ZZ(0)) || (typeof(I) <: Ideal && !iszero(I))
+            println(io, "within the vanishing set of the ideal\n", RS.defining_ideal)
+        end
+        if length(RS.inequations) > 0
+            println(io, "avoiding the zero loci of the polynomials\n", RS.inequations)
+        end
+    end
+end
+
+# Constructors
+function MatroidRealizationSpace(F::AbstractAlgebra.Ring, char::Union{Int,Nothing}, q::Union{Int,Nothing})
+    return MatroidRealizationSpace(nothing, nothing, nothing, nothing, false, F, char, q, false)
+end
+
+function MatroidRealizationSpace(I::Union{Ideal, NumFieldOrdIdl, Nothing},
+    ineqs::Union{Vector{<:Oscar.RingElem},Nothing},
+    R::Union{Oscar.MPolyRing, Ring, Nothing}, mat::Union{Oscar.MatElem,Nothing},
+    F::AbstractAlgebra.Ring, char::Union{Int,Nothing}, q::Union{Int,Nothing})
+    return MatroidRealizationSpace(I, ineqs, R, mat, nothing, F, char, q, false)
+end
+
+@doc raw"""
+is_realizable(M; char::Union{Int,Nothing}=nothing, q::Union{Int,Nothing}=nothing)
+
+* If char = nothing, then this function determines whether the matroid is realizable over some field.
+  
+* If char = 0,then this function determines whether the matroid is realizable over some field of
+    characteristic 0.
+  
+* If char = p is prime, this function determines whether the matroid is realizable
+    of characteristic p.
+  
+* if char = p and q is a power of p, this function determines whether the matroid is realizable over the
+    finite field ``GF(q)}``.
+"""
+function is_realizable(M::Matroid; char::Union{Int,Nothing}=nothing, q::Union{Int,Nothing}=nothing)::Bool
+    RS = realization_space(M, char=char, q=q)
+    return is_realizable(RS)
+end
+
+function is_realizable(RS::MatroidRealizationSpace)
+    !isnothing(RS.realizable) && return RS.realizable
+
+    if !(typeof(RS.ambient_ring) <: MPolyRing)
+        RS.realizable = true
+        return RS.realizable
+    end
+    for p in minimal_primes(RS.defining_ideal)
+        component_non_trivial = true
+        for f in RS.inequations
+            if f in p
+                component_non_trivial = false
+                break
+            end
+        end
+        if component_non_trivial
+            RS.realizable = true
+            return RS.realizable
+        end
+    end
+    RS.realizable = false
+    return RS.realizable
+end
+
+
+@doc raw"""
+defining_ideal(RS::MatroidRealizationSpace)
```

Comment:
```suggestion
    defining_ideal(RS::MatroidRealizationSpace)
```
Four spaces indentation here. Applies to many other docstrings, too.